src/Simplex/Chat.hs

```diff
@@ -4877,32 +4880,62 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
                   | otherwise -> Nothing
            in setPreference_ SCFTimedMessages ctUserTMPref' ctUserPrefs
+        createProfileUpdateItem c' =
+          when anythingExcludingPrefsChanged $ do
+            let ciContent = CIRcvDirectEvent $ RDEUpdatedProfile p p'
+            createInternalChatItem user (CDDirectRcv c') ciContent Nothing
+          where
+            anythingExcludingPrefsChanged =
+              nn /= on || nfn /= ofn || ni /= oi || ncl /= ocl
+            Profile {displayName = on, fullName = ofn, image = oi, contactLink = ocl} = p
+            Profile {displayName = nn, fullName = nfn, image = ni, contactLink = ncl} = p'
 
     xInfoMember :: GroupInfo -> GroupMember -> Profile -> m ()
-    xInfoMember gInfo m p' = void $ processMemberProfileUpdate gInfo m p'
+    xInfoMember gInfo m p' = void $ processMemberProfileUpdate gInfo m p' True
 
     xGrpLinkMem :: GroupInfo -> GroupMember -> Connection -> Profile -> m ()
     xGrpLinkMem gInfo@GroupInfo {membership} m@GroupMember {groupMemberId, memberCategory} Connection {viaGroupLink} p' = do
       xGrpLinkMemReceived <- withStore $ \db -> getXGrpLinkMemReceived db groupMemberId
       if viaGroupLink && isNothing (memberContactId m) && memberCategory == GCHostMember && not xGrpLinkMemReceived
         then do
-          m' <- processMemberProfileUpdate gInfo m p'
+          m' <- processMemberProfileUpdate gInfo m p' False
           withStore' $ \db -> setXGrpLinkMemReceived db groupMemberId True
           let connectedIncognito = memberIncognito membership
           probeMatchingMemberContact m' connectedIncognito
         else messageError "x.grp.link.mem error: invalid group link host profile update"
 
-    processMemberProfileUpdate :: GroupInfo -> GroupMember -> Profile -> m GroupMember
-    processMemberProfileUpdate gInfo m@GroupMember {memberContactId} p' =
-      case memberContactId of
-        Nothing -> do
-          m' <- withStore $ \db -> updateMemberProfile db user m p'
-          toView $ CRGroupMemberUpdated user gInfo m m'
-          pure m'
-        Just mContactId -> do
-          mCt <- withStore $ \db -> getContact db user mContactId
-          Contact {profile} <- processContactProfileUpdate mCt p' True
-          pure m {memberProfile = profile}
+    processMemberProfileUpdate :: GroupInfo -> GroupMember -> Profile -> Bool -> m GroupMember
+    processMemberProfileUpdate gInfo m@GroupMember {memberProfile = p, memberContactId} p' createItems
+      | onlyMemberProfileFields (fromLocalProfile p) /= onlyMemberProfileFields p' =
+          case memberContactId of
+            Nothing -> do
+              m' <- withStore $ \db -> updateMemberProfile db user m p'
+              createProfileUpdateItem m'
+              toView $ CRGroupMemberUpdated user gInfo m m'
+              pure m'
+            Just mContactId -> do
+              mCt <- withStore $ \db -> getContact db user mContactId
+              if canUpdateProfile mCt
```

Comment:
Isn't getContact failable, and it will simply throw exception if there is no contact?

So you probably need to conditionally read it outside of case memberContactId?